src/ray/common/ray_object.cc

```diff
@@ -33,17 +89,22 @@ std::shared_ptr<LocalMemoryBuffer> MakeErrorMetadataBuffer(rpc::ErrorType error_
   return MakeBufferFromString(meta);
 }
 
-RayObject::RayObject(rpc::ErrorType error_type)
-    : RayObject(nullptr, MakeErrorMetadataBuffer(error_type), {}) {}
-
-RayObject::RayObject(rpc::ErrorType error_type, const std::string &append_data)
-    : RayObject(MakeBufferFromString(append_data), MakeErrorMetadataBuffer(error_type),
-                {}) {}
+RayObject::RayObject(rpc::ErrorType error_type, const rpc::RayErrorInfo *ray_error_info) {
+  if (ray_error_info == nullptr) {
+    Init(nullptr, MakeErrorMetadataBuffer(error_type), {});
+    return;
+  }
 
-RayObject::RayObject(rpc::ErrorType error_type, const uint8_t *append_data,
-                     size_t append_data_size)
-    : RayObject(MakeBufferFromString(append_data, append_data_size),
-                MakeErrorMetadataBuffer(error_type), {}) {}
+  RAY_CHECK(ray_error_info->has_actor_init_failure());
+  // This is temporarily here because changing this requires changes in all language
+  // frontend.
+  // TODO(sang, lixin): Remove it.
+  const auto final_buffer =
+      SerializeErrorInfo<rpc::RayException>(ray_error_info->actor_init_failure());
+  Init(MakeBufferFromString(final_buffer->Data(), final_buffer->Size()),
```

Comment:
final_buffer does look like a buffer already, why we call MakeBufferFromString?

Reply:
Hmm good point. This was refactoring. Let me think about better names.